packages/commonwealth/client/scripts/views/Layout.tsx

```diff
@@ -45,9 +45,17 @@ const ApplicationError = () => {
 
 interface ShouldDeferChainAttrs {
   deferChain: boolean;
+  scope?: string;
 }
 
-const shouldDeferChainLoading = ({ deferChain }: ShouldDeferChainAttrs) => {
+const shouldDeferChainLoading = ({
+  deferChain,
+  scope,
+}: ShouldDeferChainAttrs) => {
+  // with quick-switcher, sometimes this renders before the correct chain is selected,
+  // so we must check if they match. Defer chain load if they do not match yet.
+  if (scope && scope !== app?.chain?.id) return true;
```

Comment:
This condition I've only noticed with quick switcher. See video for demo:

https://www.loom.com/share/3023c87a2ca746e0826f5e7b967b676b

But good point on non-deferred pages. I found one case where it did not load (a specific proposal page), so I will find a different way to handle this.

Reply:
@mhagel please mark as draft until fix is ready to go.

Reply:
Now avoiding loading chain before redirect completes by creating a second check for the community home path.
There is a moment where /osmosis starts loading before the redirect route is rendered, so the `deferChain` flag isn't noticed by `Layout`.

Now, by checking if the location pathname matches the route param scope exactly, we can detect if we are on the /osmosis URL. We avoid 2 possible initChain paths by using this check. (Both were running otherwise).